File: 1760_minimum_limit_of_balls_in_a_bag.java
class Solution {
    public int minimumSize(int[] p, int q) {
        int s = 1, e = Arrays.stream(p).max().getAsInt();
        while (s < e) {
            int m = (s + e) >> 1, v = 0;
            for (int k : p) v += (k - 1) / m;
            if (v > q) s = m + 1; else e = m;
        }
        return s;
    }
}

File: 1760_minimum-limit-of-balls-in-a-bag.cpp
class Solution {
public:
    int minimumSize(vector<int>& g, int o) {
        int x = 1, y = *max_element(g.begin(), g.end());
        while (x < y) {
            int m = (x + y) >> 1, f = 0;
            for (auto t : g) f += (t - 1) / m;
            f > o ? x = m + 1 : y = m;
        }
        return x;
    }
};
